gems/smithy-model/lib/smithy-model/shapes.rb

```diff
@@ -207,19 +221,22 @@ def initialize(options = {})
         # @return [Class]
         attr_accessor :type
 
-        # @return [MemberShape]
-        def add_member(name, shape, traits: {})
-          @members[name] = MemberShape.new(shape, traits: traits)
+        def add_member(name, member_name, shape, traits: {})
+          @members_by_name[member_name] = name
+          @members[name] = MemberShape.new(member_name, shape, traits: traits)
         end
 
+        # @param [Symbol, String] name
         # @return [Boolean]
         def member?(name)
-          @members.key?(name)
+          @members.key?(name) || @members_by_name.key?(name)
```

Comment:
I want flexibility to check members existence whether by the symbolized member name (`:city_id`) vs the member name defined on the shape (`cityName`). I also did this for `member(name)` method

Reply:
I'm not sure if we want to do that. We may get something returned unexpectedly?